src/extension.ts

```diff
@@ -54,15 +54,30 @@ class RunOnSaveExtension {
 
     const startMs = performance.now();
     let pendingCount = cmds.length;
+    let hasError = false;
 
-    const onCmdComplete = (cfg: ICommand, elapsedMs: number) => {
+    const onCmdComplete = (cfg: ICommand, res: IExecResult) => {
       --pendingCount;
       this.showOutputMessageIfDefined(cfg.messageAfter);
       this.showOutputMessageIfDefined(
-        cfg.showElapsed && `Elapsed ms: ${elapsedMs}`,
+        cfg.showElapsed && `Elapsed ms: ${res.elapsedMs}`,
       );
 
+      if (cfg.autoShowOutputPanel === 'error' && !hasError) {
+        // sets flag when the first command error occurs
+        hasError = res.statusCode !== 0;
+        if (hasError) {
+          // shows the output panel without changing focus
+          this._outputChannel.show(true);
+        }
+      }
+
       if (pendingCount === 0) {
+        // hides the output panel on a successful save iteration
+        if (cfg.autoShowOutputPanel === 'error' && this.autoClearConsole && !hasError) {
```

Comment:
I'm hesitant to auto hide since we don't know for sure if the user explicitly displayed the panel or if it was shown automatically. It might be that a user wants the console to clear on every save to get fresh output but doesn't necessarily want the panel to disappear.